test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -10118,6 +10107,80 @@ SimpleTestPackageContext createTestPackage(string name, string version, List<Sim
             }
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        [InlineData(null)]
+        public async Task RestoreNetCore_CPVMProject_ManagePackageVersionsCentrally_CanBeDisabled(bool? managePackageVersionsCentrally)
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+                var packagesForSource = new List<SimpleTestPackageContext>();
+                var packagesForProject = new List<SimpleTestPackageContext>();
+                var framework = FrameworkConstants.CommonFrameworks.NetCoreApp20;
+
+                SimpleTestPackageContext createTestPackage(string name, string version, List<SimpleTestPackageContext> source)
+                {
+                    var result = new SimpleTestPackageContext()
+                    {
+                        Id = name,
+                        Version = version
+                    };
+                    result.Files.Clear();
+                    source.Add(result);
+                    return result;
+                };
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                   "projectA",
+                   pathContext.SolutionRoot,
+                   framework);
+
+                // the package references defined in the project should not have version unless CPM is disabled
+                var packageBNoVersion = createTestPackage("B", managePackageVersionsCentrally is null || managePackageVersionsCentrally == true ? null : "1.0.0", packagesForProject);
+                var packageB100 = createTestPackage("B", "1.0.0", packagesForSource);
+                var packageC100 = createTestPackage("C", "1.0.0", packagesForSource);
+                var packageC200 = createTestPackage("C", "2.0.0", packagesForSource);
```

Comment:
Are you needing this one in the array, `packagesForSource`? It'd be easier for me to understand if the variable is removed if it's unused.

```suggestion
                createTestPackage("C", "2.0.0", packagesForSource);
```

Reply:
This whole test is just a copy/paste so I'm not as worried about poor patterns, do you mind yourself fixing up test classes when you work on a new test?

Reply:
>  do you mind yourself fixing up test classes when you work on a new test?

When I write **new tests**, I try to follow our guidelines. If it means something complicated like creating a bunch of helper methods, I may or may not do that. If it's something trivial like using proper variable names or conventions, I tend to do that in my new tests. I don't tend to update entire class files (which is not what I suggested, to be clear).